src/app/GitUI/Editor/Diff/DiffHighlightService.cs

```diff
@@ -148,38 +129,59 @@ private void SetText(ref string text)
         text = sb.ToString();
     }
 
+    /// <summary>
+    /// Set highlighting for <paramref name="text"/>.
+    /// The parsed added/removed lines in <see cref="_diffLinesInfo"/> is used as well as
+    /// the highlighting in <see cref="_textMarkers"/> (if Git highlighting <see cref="_useGitColoring"/>),
+    /// is used to mark inline differences (dim unchanged part of lines).
+    /// </summary>
+    /// <param name="text">The text to process.</param>
+    internal void SetHighlighting(string text)
+    {
+        if (!_useGitColoring)
+        {
+            HighlightAddedAndDeletedLines(_textMarkers);
+        }
+
+        // Apply GE word highlighting for Patch display (may apply to Difftastic setting, if not available for a repo)
+        if (!_useGitColoring || AppSettings.DiffDisplayAppearance.Value != GitCommands.Settings.DiffDisplayAppearance.GitWordDiff)
+        {
+            MarkInlineDifferences(text, _textMarkers);
+        }
+    }
+
     /// <summary>
     /// Highlight lines that are added, removed and header lines.
     /// This is an alternative configuration to use the Git diff coloring (that has more features).
     /// </summary>
-    private void HighlightAddedAndDeletedLines(IDocument document)
+    /// <param name="textMarkers">The markers to append to.</param>
+    private void HighlightAddedAndDeletedLines(List<TextMarker> textMarkers)
     {
         foreach (ISegment segment in GetAllLines(DiffLineType.Minus))
         {
-            document.MarkerStrategy.AddMarker(CreateTextMarker(segment.Offset, segment.Length, _removedBackColor));
+            textMarkers.Add(CreateTextMarker(segment.Offset, segment.Length, _removedBackColor));
```

Comment:
```suggestion
            textMarkers.Insert(0, CreateTextMarker(segment.Offset, segment.Length, _removedBackColor));
```
in the entire function